File: app/src/main/java/org/michaelbel/moviemade/ui/modules/main/ResultsMvp.java
package org.michaelbel.moviemade.ui.modules.main;

import com.arellomobile.mvp.MvpView;

import org.michaelbel.moviemade.annotation.EmptyViewMode;
import org.michaelbel.moviemade.data.dao.Movie;

import java.util.List;

public interface ResultsMvp extends MvpView {

    void showResults(List<Movie> results, boolean firstPage);

    void showError(@EmptyViewMode int mode);
}

File: app/src/main/java/org/michaelbel/moviemade/ui/modules/main/ResultsMvp.kt
package org.michaelbel.moviemade.ui.modules.main

import com.arellomobile.mvp.MvpView

import org.michaelbel.moviemade.annotation.EmptyViewMode
import org.michaelbel.moviemade.data.dao.Movie

interface ResultsMvp : MvpView {

    fun setResults(results: List<Movie>, firstPage: Boolean)

    fun setError(@EmptyViewMode mode: Int)
}